python/cudf/cudf/core/tools/numeric.py

```diff
@@ -20,6 +20,17 @@
 from cudf.core.column import as_column
 from cudf.utils.dtypes import can_convert_to_column
 
+DOWNCAST_TYPE_MAP = {
+    "integer": list(np.typecodes["Integer"]),
+    "signed": list(np.typecodes["Integer"]),
+    "unsigned": list(np.typecodes["UnsignedInteger"]),
+}
```

Comment:
Rather than storing this list of characters we could store `cudf.dtype(t) for t in np.typecodes[...]` so that the downcast code doesn't have to call it every time. If we really want to be fancy we could populate the dict lazily to avoid the up front cost if a user never calls this. Up to you to decide if any of this is worth it though; I don't know if the `to_numeric` function is all that important and perhaps optimizing for import time and less development effort is best here.

Reply:
Yeah, I think we can re-work this logic a bit more, let me get back with new code

Reply:
Changed the logic.